oscore-cf/src/main/java/org/eclipse/californium/oscore/Decryptor.java

```diff
@@ -195,7 +201,16 @@ protected static Encrypt0Message prepareCOSEStructure(byte[] protectedData) {
 	protected static Encrypt0Message decompression(byte[] cipherText, Message message) throws OSException {
 		Encrypt0Message enc = new Encrypt0Message(false, true);
 
-		decodeObjectSecurity(message, enc);
+		//Added try-catch for general Exception. The array manipulation can cause exceptions.
+		try {
+			decodeObjectSecurity(message, enc);
+		} catch (OSException e) {
+			LOGGER.error(e.getMessage());
+			throw new OSException(e.getMessage());
```

Comment:
Why is the OSException not simply thrown again?
`throw e` would keep the original stacktrace.

Reply:
I will change it accordingly.